pandas/core/window/rolling.py

```diff
@@ -247,8 +247,31 @@ def __repr__(self) -> str:
         return f"{self._window_type} [{attrs}]"
 
     def __iter__(self):
-        url = "https://github.com/pandas-dev/pandas/issues/11704"
-        raise NotImplementedError(f"See issue #11704 {url}")
+        window = self._get_window(win_type=None)
+
+        blocks, obj = self._create_blocks()
+        index = self._get_window_indexer(window=window)
+
+        # Choose the min between min_periods and window to determine the output size
+        if self.min_periods is None:
+            iter_threshold = window
+        else:
+            iter_threshold = min(window, self.min_periods)
+
+        start, end = index.get_window_bounds(
+            num_values=len(obj),
+            min_periods=self.min_periods,
+            center=self.center,
+            closed=self.closed,
+        )
+        # From get_window_bounds, those two should be equal in length of array
+        assert len(start) == len(end)
+
+        window_size = len(start)
+        for i in range(window_size):
+            result = obj.iloc[slice(start[i], end[i])]
+            if result.count().min() >= iter_threshold:
```

Comment:
IMO since `min_periods` is supposed to influence the the aggregation result, and we're just returning the window here, I think we should always return the result and not filter results based on `iter_threshold`

Reply:
emm, i am not sure about this actually, because `df.rolling(window=2, min_periods=3).sum()` for instance is not allowed, and will raise a `ValueError` because min_periods has to be equal or smaller than window in aggregation, that is why i think maybe it makes a bit more sense to have the minimum here between `window` and `min_periods`.

but I also do not know if this matters in `iter`? maybe raise an error as well?

Reply:
Right I am thinking it doesn't matter in `__iter__` because the aggregation hasn't happened yet (`sum`), and it's up to the user to decide what they want to do with each window.

From the user perspective, I can see a potential source of confusion if not all the windows are returned.

Reply:
yeah, leaving it to users sounds more convincing!

thanks, @mroeschke ! i have updated the PR to remove this check (also not error raising) here.

Reply:
Looks good, thanks! Might want to document this behavior in `computation.rst`

Reply:
sure, added! thanks for the reviews! @mroeschke